models/staging/stg_sessions.sql

```diff
@@ -12,16 +12,25 @@
 WITH
 window_functions as (
     select i.*,
+    --story
       first_value(story_intent ignore nulls) over
         (partition by session_id
          order by interaction_initiation_timestamp
          ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING
          ) as first_story_intent,
+         -- model
            first_value(interaction_model_id ignore nulls) over
         (partition by session_id
          order by interaction_initiation_timestamp
          ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING
-         ) as first_bot_model
+         ) as first_bot_model,
+         -- env
+           first_value(environment ignore nulls) over
+        (partition by session_id
+         order by interaction_initiation_timestamp
+         ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING
+         ) as first_bot_environment,
```

Comment:
I would just name it environment, it is really impossible to have the same session on two envs. it's a bug if that happens. OFC the query that takes first env is good but name does not need to be so pedantic